torch/_inductor/codegen/cpp.py

```diff
@@ -785,14 +789,25 @@ def load(self, name: str, index: sympy.Expr):
         new_index = self.transform_index(index)
 
         if expanded_index == new_index:
-            line = f"at::vec::Vectorized<float>({var}[{cexpr(index)}])"
+            if V.graph.get_dtype(name) in [torch.bool, torch.uint8, torch.float64]:
+                g_tmp_buf = f"g_tmp_buffer_{var}"
+                nelements = 1
+                if f"float {g_tmp_buf}[{nelements}] = {{0}};" not in self.loads._lines:
```

Comment:
Please explain why we need this check.

Reply:
Please see https://github.com/pytorch/pytorch/pull/90270, if one bool/uint8 tensor is used by many times, it will call many time load.

Reply:
So, you are going to fix #90270 in this PR too?

Reply:
Yes, i am meet the same issue for a the test case after enabling cast float.